app/javascript/admin/controllers/config_controller.js

```diff
@@ -246,4 +259,68 @@ export default class ConfigController extends Controller {
     }
   }
   // AUTH PROVIDERS FUNCTIONS END
+
+  enabledProvidersWithMissingKeys() {
+    const providersWithMissingKeys = [];
+    document
+      .querySelectorAll('[data-enable-auth="true"]')
+      .forEach((provider) => {
+        const providerName = provider.dataset.authProviderEnable;
+        const officialName = provider.dataset.authProviderOfficial;
+        if (
+          document.getElementById(`site_config_${providerName}_key`).value
+            .length === 0 ||
+          document.getElementById(`site_config_${providerName}_secret`).value
+            .length === 0
+        ) {
+          providersWithMissingKeys.push(officialName);
+        }
+      });
+
+    return providersWithMissingKeys;
+  }
+
+  generateProvidersList(providers) {
+    let list = '';
+    for (let i = 0; i < providers.length; i++) {
+      list += `<li>${providers[i]}</li>`;
+    }
+    return list;
+  }
+
+  missingKeysModalTitle() {
+    return 'Setup not complete';
+  }
```

Comment:
Is there a reason we need to make this a function? Since it is just returning a string, I think we can pass that string along directly to `adminModal`, right?